src/libs/actions/IOU.ts

```diff
@@ -1605,6 +1597,11 @@ function createSplitsAndOnyxData(
             key: `${ONYXKEYS.COLLECTION.TRANSACTION}${splitTransaction.transactionID}`,
             value: splitTransaction,
         },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${CONST.IOU.OPTIMISTIC_TRANSACTION_ID}`,
+            value: null,
+        },
```

Comment:
Clearing the draft here is executed too early; before the user finishes the split flow (before navigation) which resulted in UI briefly showing stale data

Coming from https://github.com/Expensify/App/issues/52196